### ISSUE TITLE
Incorrect issues detected for multiline f-string

### ISSUE THREAD

**Issue by 20x48** [microsoft/pylance-release#1401]
## Environment data

-   Language Server version: `v2021.6.0`
-   OS and version: `Windows 10`
-   Python version (& distribution if applicable, e.g. Anaconda): `3.8.2 (tags/v3.8.2:7b3ab59, Feb 25 2020, 23:03:10)`

## Expected behaviour

Although I know this is not a good code habit, but it can run normally. No errors should be detected.

## Actual behaviour

```
Line | Question
   3 | E: String literal is unterminated
   4 | E: Unterminated expression in f-string; missing close brace
   6 | W: "t_chunks" is not defined
   9 | E: Single close brace not allowed within f-string literal; use double close brace
```

## Code Snippet / Additional information

```python
return f'''<ul class="yhb-lst_t">{
        ''.join(
            map(lambda t_chunks: f"""<li class="_{
                '~x-:+v*'.find(t_chunks[0]) + 1
            }">{
                self.block(t_chunks[1])
            }</li>""", content)
        )
    }</ul>'''
```

**Comment by erictraut** [comment]
Thanks for reporting the problem. The parser in pylance wasn't properly handling a carriage return within an inner triple-quoted string within a an outer triple-quoted f-string. I've updated the code to handle this situation properly, and the fix will be in the next release of pylance.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.6.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202161-9-june-2021
